File: CS/LinqToSqlMasterDetail/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;

namespace LinqToSqlMasterDetail
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            NwindDataContext dc = new NwindDataContext();
            XtraReport1 report = new XtraReport1();
            BindingSource bs = (BindingSource)report.DataSource;
            bs.DataSource = dc.Orders;
            report.ShowPreviewDialog();
        }
    }
}

File: VB/LinqToSqlMasterDetail/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

Namespace LinqToSqlMasterDetail
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim dc As New NwindDataContext()
            Dim report As New XtraReport1()
            Dim bs As BindingSource = CType(report.DataSource, BindingSource)
            bs.DataSource = dc.Orders
            report.ShowPreviewDialog()
        End Sub
    End Class
End Namespace
